mechanicalsoup/stateful_browser.py

```diff
@@ -152,6 +152,24 @@ def open_relative(self, url, *args, **kwargs):
         """
         return self.open(self.absolute_url(url), *args, **kwargs)
 
+    def refresh(self):
+        """Reloads the current page with the same request as originally done.
+        Works only if page was loaded using ``StatefulBrowser`` methods.
+        The form selection and values are not retained.
+
+        :raise ValueError: Raised if no refreshable page is loaded, e.g., when using
+            the shallow ``Browser`` wrapper functions.
+
+        :return: Response of the request."""
+        old_request = self.__state.request
+        if old_request is None:
+            raise ValueError('The current page is not refreshable')
```

Comment:
A hint on why this fails would be welcome. But we can also merge this as-is and improve the message later if needed.

Reply:
I am not sure about this one. I extended the description, but in the end we have something similar to the not-working hint above.